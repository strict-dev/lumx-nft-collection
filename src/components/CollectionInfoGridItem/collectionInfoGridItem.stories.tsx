import { ComponentMeta, ComponentStory } from '@storybook/react'

import CollectionInfoGridItem from '.'

export default {
  title: 'CollectionInfoGridItem',
  component: CollectionInfoGridItem
} as ComponentMeta<typeof CollectionInfoGridItem>

const Template: ComponentStory<typeof CollectionInfoGridItem> = (args) => (
  <CollectionInfoGridItem {...args} />
)

export const Default = Template.bind({})

Default.args = {
  type: 'vol',
  title: 'Volume',
  data: '1,000'
}
