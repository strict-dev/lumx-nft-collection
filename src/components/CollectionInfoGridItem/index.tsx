import MenuIcon from '@public/assets/menu-1.svg'
import Image from 'next/image'

type CollectionInfoGridItem = {
  title: string
  data: string
  children?: React.ReactNode
  type?: 'vol'
}

const CollectionInfoGridItem = ({
  title,
  data,
  children,
  type
}: CollectionInfoGridItem) => {
  return (
    <div className="flex flex-col items-center justify-center gap-1 box">
      <p className="text-xs font-bold text-neutral-600">
        {title.toUpperCase()}
      </p>
      <div className="flex items-center gap-1">
        {type === 'vol' && (
          <Image src={MenuIcon} alt="Icon" width={16} height={16} />
        )}
        <span className="text-lg font-bold">{data}</span>
      </div>
      {children}
    </div>
  )
}

export default CollectionInfoGridItem
