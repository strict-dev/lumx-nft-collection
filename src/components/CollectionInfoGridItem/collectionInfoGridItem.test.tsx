import { render } from '@testing-library/react'

import CollectionInfoGridItem from '.'

describe('<CollectionInfoGridItem />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <CollectionInfoGridItem data="1000" title="Vol" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
