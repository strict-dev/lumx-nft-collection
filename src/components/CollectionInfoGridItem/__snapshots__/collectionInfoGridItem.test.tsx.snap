// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionInfoGridItem /> should render the heading 1`] = `
<div
  class="flex flex-col items-center justify-center gap-1 box"
>
  <p
    class="text-xs font-bold text-neutral-600"
  >
    VOL
  </p>
  <div
    class="flex items-center gap-1"
  >
    <span
      class="text-lg font-bold"
    >
      1000
    </span>
  </div>
</div>
`;
